CorsixTH/Lua/hospitals/player_hospital.lua

```diff
@@ -536,6 +536,49 @@ function PlayerHospital:makeRaiseRequest(amount, staff)
   self.world.ui.bottom_panel:queueMessage("strike", amount, staff)
 end
 
+--! Announce to the player that a staff member is leaving the hospital
+--!param staff (table) The staff member
+function PlayerHospital:announceStaffLeave(staff)
+  self.world.ui:playRandomAnnouncement(staff.leave_sounds, staff.leave_priority)
+end
+
+--! Advise the player that they attempted to put a staff member in a room they can't attend to
+--!param room (table)
+--!param staff_member (table)
+function PlayerHospital:adviseWrongStaffForRoom(room, staff_member)
+  local room_name = room.room_info.long_name
+  local required = room.room_info.maximum_staff or room.room_info.required_staff
+  if staff_member.humanoid_class == "Nurse" then
+    self.world.ui.adviser:say(_A.staff_place_advice.nurses_cannot_work_in_room:format(room_name))
+  elseif required then
+    if required.Nurse then
+      self.world.ui.adviser:say(_A.staff_place_advice.only_nurses_in_room:format(room_name))
+    elseif required.Surgeon then
+      self.world.ui.adviser:say(_A.staff_place_advice.only_surgeons)
+    elseif required.Researcher then
+      self.world.ui.adviser:say(_A.staff_place_advice.only_researchers)
+    elseif required.Psychiatrist then
+      self.world.ui.adviser:say(_A.staff_place_advice.only_psychiatrists)
+    else
+      self.world.ui.adviser:say(_A.staff_place_advice.only_doctors_in_room:format(room_name))
+    end
+  else
+    self.world.ui.adviser:say(_A.staff_place_advice.doctors_cannot_work_in_room:format(room_name))
+  end
+end
+
+--! Advise the player that a multi staff occupancy room has more staff than desks
+--!param staff_member (table)
+--!param room_id (string) The id from room_info
+function PlayerHospital:adviseNeedsDesk(staff_member, room_id)
+  if staff_member.profile.is_researcher and room_id == "research" then
```

Comment:
I don't love moving this logic here. It's turning hospitial into a god class like world. Isn't calling hospital:giveAdvice from room enough to do what you're trying to do?